flexget/plugins/clients/qbittorrent.py

```diff
@@ -120,9 +126,76 @@ def connect(self, config):
         logger.debug('Successfully connected to qBittorrent')
         self.connected = True
 
+    def check_torrent_exists(self, file_path, verify_cert):
+        if not self.connected:
+            raise plugin.PluginError('Not connected.')
+
+        hash_torrent = None
+
+        if 'magnet:' in file_path:
+            hash_search = re.search('magnet\:.*btih:((\w*))', file_path, re.IGNORECASE)
+            if hash_search:
+                hash_torrent = hash_search.group(1)
+        else:
+            try:
+                torrent = Torrent.from_file(file_path)
+            except FileNotFoundError as e:
+                logger.error('Error checking torrent file, file {} does not exist', file_path)
+                return False
+
+            hash_torrent = torrent.info_hash
```

Comment:
We already grab this info from [magnets](https://github.com/Flexget/Flexget/blob/e3df823b889571f844a42ef670be965399c23929/flexget/components/bittorrent/magnet_info_hash.py#L28) and [torrent files](https://github.com/Flexget/Flexget/blob/e3df823b889571f844a42ef670be965399c23929/flexget/components/bittorrent/torrent.py#L63), and put it in the `torrent_info_hash` entry field. I think it would be better to just pull the hash from there rather than repeat that logic.

Reply:
@gazpachoking Done